src/Controller/Controller.java

```diff
@@ -0,0 +1,51 @@
+package Controller;
+
+import com.sun.istack.internal.NotNull;
+
+import java.util.Map;
+
+import Interactor.Analyzer;
+import Interactor.Report;
+import Interactor.RequestModel;
+import Presenters.HTMLPresenter.HTMLPresenter;
+import Presenters.PDFPresenter.PDFPresenter;
+
+/**
+ *
+ */
+public class Controller {
+
+    private final RequestModel model;
+    private final Report report;
+    private Presenter presenter;
+
+    /**
+     * Default constructor
+     */
+    public Controller(final RequestModel model, final Analyzer analyzer) {
+        this.model = model;
+        this.report = analyzer.analyze();
```

Comment:
What if this operation is long-running ?
Can we analyze directly in constructor ?
Please rewrite this to analyze data only when we request the report in show method